pkg/volume/vsphere_volume/vsphere_volume.go

```diff
@@ -391,6 +393,10 @@ func (v *vsphereVolumeProvisioner) Provision() (*v1.PersistentVolume, error) {
 		pv.Spec.AccessModes = v.plugin.GetAccessModes()
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.BlockVolume) {
+		pv.Spec.VolumeMode = v.options.PVC.Spec.VolumeMode
```

Comment:
I'm afraid that vSphere doesn't seem to support BlockVolume, so shouldn't it return error here?

Reply:
Fixed, now it returns the same error as other plugins that don't support block.